pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -440,12 +453,81 @@ func (m *cgroupManagerImpl) Update(cgroupConfig *CgroupConfig) error {
 		libcontainerCgroupConfig.PidsLimit = *cgroupConfig.ResourceParameters.PidsLimit
 	}
 
+	if m.adapter.cgroupManagerType == libcontainerSystemd {
+		if err := m.updateSlicePropertiesSystemd(libcontainerCgroupConfig); err != nil {
+			return fmt.Errorf("failed to slice properties for unit %v: %v", cgroupConfig.Name, err)
+		}
+	}
 	if err := setSupportedSubsystems(libcontainerCgroupConfig); err != nil {
 		return fmt.Errorf("failed to set supported cgroup subsystems for cgroup %v: %v", cgroupConfig.Name, err)
 	}
 	return nil
 }
 
+// Since libcontainer doesn't support updating slice properties, and only sets resources params during
+// creating of slice, we have to update them manually.
+//
+// The code is adapted from:
+// https://github.com/opencontainers/runc/blob/v1.0.0-rc10/libcontainer/cgroups/systemd/apply_systemd.go#L138
+func (m *cgroupManagerImpl) updateSlicePropertiesSystemd(c *libcontainerconfigs.Cgroup) error {
+	var (
+		unitName   = c.Name
+		properties []systemddbus.Property
+	)
+
+	if c.Resources.Memory != 0 {
+		properties = append(properties,
+			newProp("MemoryLimit", uint64(c.Resources.Memory)))
+	}
+
+	if c.Resources.CpuShares != 0 {
```

Comment:
this won't work on cgroup v2.

We would need to set the `CPUWeight=` property (https://github.com/kubernetes/kubernetes/pull/89567 moves the conversion to a new function so should be easier to use)